### ISSUE TITLE
Various test failures

### ISSUE THREAD

**Issue by eserte** [KalessinD/perl_mojox_http_async#1]
I see various test failures on my smoker systems:
* t/insecure.t may fail like this
```
#   Failed test 'checking the response message'
#   at t/insecure.t line 154.
#          got: 'Connection reset by peer'
#     expected: 'Request timeout'

#   Failed test 'checking the response code'
#   at t/insecure.t line 155.
#          got: '520'
#     expected: '524'
# Looks like you failed 2 tests of 35.
t/insecure.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/35 subtests 
```
* t/secure.t may fail like this
```
Can't call method "accept" on an undefined value at t/secure.t line 54.
cannot open port: localhost:46095 at /home/cpansand/.cpan/build/2021112623/Test-TCP-2.22-0/blib/lib/Test/TCP.pm line 53.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 111 just after 1.
t/secure.t .... 
Dubious, test returned 111 (wstat 28416, 0x6f00)
All 1 subtests passed 
```
or
```
connect error: Connection refused at t/secure.t line 127.
 at t/secure.t line 127.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 61 just after 1.
t/secure.t .... 
Dubious, test returned 61 (wstat 15616, 0x3d00)
All 1 subtests passed 
```

I don't see any pattern regarding OS or perl versions.

**Comment by KalessinD** [comment]
Thank you for report!

I've just uploaded some fixes for "t/insecure.t" test file.

And i'm investigating the issue with "t/secure.t" on some platforms.

So on Ubuntu i can't reproduce this failure. But i see the same issue on Cent OS.
And it seems that something wrong happens in  "IO::Socket::SSL" package while processing "SSL_cert_file" option.

This command doesnt' fail:

```bash
$ perl -M5.026 -MIO::Socket::SSL -Mwarnings -e 'my $s = IO::Socket::SSL->new(LocalAddr => "localhost", LocalPort => 40516, Listen => 3, SSL_key_file => "t/certs/server-key.pem") or die $!;'
```

But the same comamnd as above, but with "SSL_cert_file" option - fails:
```bash
$ perl -M5.026 -MIO::Socket::SSL -Mwarnings -e 'my $s = IO::Socket::SSL->new(LocalAddr => "localhost", LocalPort => 40516, Listen => 3, SSL_key_file => "t/certs/server-key.pem", SSL_cert_file => "t/certs/server-cert.pem") or die $!;'
Inappropriate ioctl for device at -e line 1
```

P.S.

Seems that Cent OS has a different behaviour with OpenSSL.

`Net::SSLeay::CTX_use_certificate_chain_file($ctx, "t/certs/server-cert.pem")` doesn't see the PEMcertificate.
```
[format:PEM] error:140AB18F:SSL routines:SSL_CTX_use_certificate:ee key too small
```

**Comment by eserte** [comment]
Maybe it depends on the installed openssl version? On my systems, your oneliner fails for example on CentOS 8, Fedora and Ubuntu 20.04, but not on CentOS 7 and Ubuntu 18.04.

**Comment by KalessinD** [comment]
I've found on stackoverflow that some new systems requires 2048Kb key size.
I've re-generate a server certificate on Cent OS.
And now it works fine for me. 

Let's see the the result for 0.06 version. ^)

Thank you for report and your assumption :)

**Comment by KalessinD** [comment]
Seems that all issues with *nix systems were fixed in v0.07

![image](https://user-images.githubusercontent.com/67797273/143885402-2467678c-a9c5-4598-838a-4f6cdd9a1e0e.png)

Closing the issue.

Don't hesitate to re-open if don't agree.
